### ISSUE TITLE
Mac Shift Scroll

### ISSUE THREAD

**Issue by desagas** [MinecraftForge/MinecraftForge#7557]
Mac users can not use SHIFT and MOUSE SCROLL at the same time. As many great mods use this feature of shifting and scrolling, like Dank and Tank Storage, and FTB Utilmine, etc, Mac users are not able to use these mods. In 1.13, as all mac users know and hate, Mojang update to LWJGL3, which completely cause this functionality to disappear for all mac users. Ugh. I was hoping Forge was planning on implementing this in a future update/release.

Anyways, thank god for keybinding remapping software and control selection in Minecraft.

**Comment by desagas** [comment]
https://github.com/andyvanee/hscroll/blob/master/src/main/java/net/fabricmc/andyvanee/mixin/MinecraftClientMixin.java

**Comment by LexManos** [comment]
Interesting, I didn't know macs had a horizontal scroll feature. Seems like whomever at Mojang that re-wrote the mouse handler didn't either. So did you report this to vanilla as a bug?

**Comment by AterAnimAvis** [comment]
Seems there's an open report, https://bugs.mojang.com/browse/MC-121772

**Comment by desagas** [comment]
Yes, it’s been an issue for a while. But, Mojang refuses to fix it, and they are now deleting any further discussions about it. I figure, in vanilla, it’s not so much a big deal, but in modded, it can be.

**Comment by desagas** [comment]
Ive been looking into the code for a couple of hours now, and I have found where the mouse events are happening, but am struggling to figure out how/where `scrollDelta` is calculated. Even as I go further into _MouseHelper_, I can see `accumulatedscrollDelta`, but I still can not not find where it is calculated. If I could find that, I can quickly try and code something up for a PR review. Hmmm ... ill try again later.

**Comment by TheCurle** [comment]
This appears to be a LWJGL issue.
TLDR: LWJGL2 did mouseScroll(dY != 0 ? dY : dX), LWJGL3 does mouseScroll(dX, dY).
This is a somewhat trivial fix, but it requires a rewrite of a sizeable chunk of input handle.
See more information here: https://bugs.mojang.com/browse/MC-121772?focusedCommentId=534520&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-534520

**Comment by desagas** [comment]
Would they not be able to simply add the Y and X offsets together, from the mouse scrolling capture events, as they work fine, since Minecraft does not use horizontal scrolling?

**Comment by desagas** [comment]
@LexManos I think I have a potentially super quick fix, but I would like to test it first, to make sure I'm not an idiot. Is it possible to edit and rebuild the SNAPSHOT files to test PR ideas for Forge, like other mods, in IntelliJ? Also, is it possible to make PRs for MINECRAFT PATCH code, as they are not full files. If so, can you please tell me what the four numbers inbetween the `@@ -# #  +# #` symbols represent?

**Comment by desagas** [comment]
I have fixed and tested it in forge..38, and it works fine. This is my PR: https://github.com/MinecraftForge/MinecraftForge/pull/7592

**Comment by TheCurle** [comment]
This issue still persists. If someone (that preferably knows how an IDE and git works) wants to take a crack at this, this is an excellent first contribution.
